lib/ttcp

```diff
@@ -191,21 +214,144 @@ __ttcp::spin () {
     _TTCP_SPIN_PID=$!
 }
 
-__ttcp::check_env () {
+__ttcp::generate_credentials () {
+    TTCP_ID_CLIP=${TTCP_ID_CLIP:-"$(echo "${TTCP_ID}${TTCP_SALT1}" | __ttcp::hash)"}
+    TTCP_PASSWORD_CLIP=${TTCP_PASSWORD_CLIP:-"$(echo "${TTCP_PASSWORD}${TTCP_SALT1}" | __ttcp::hash)"}
+    TTCP_ID_TRANS=${TTCP_ID_TRANS:-"$(echo "${TTCP_ID}${TTCP_SALT2}" | __ttcp::hash)"}
+    TTCP_PASSWORD_TRANS=${TTCP_PASSWORD_TRANS:-"$(echo "${TTCP_PASSWORD}${TTCP_ID}${TTCP_SALT2}" | __ttcp::hash)"}
+}
+
+__ttcp::init () {
+    # Stop immediately in case of any errors.
+    # Because this function may make strange changes.
+    set -e
+
+    # Get a filename calling the function.
+    # http://stackoverflow.com/a/192319/
+    local _file="${0##*/}"
+    # Remove file's extention
+    local _cmd="${_file%.*}"
+
+    local _given_id=""
+    local _given_password1=""
+    local _given_password2=""
+
+    # If there is any standard input, discard it.
+    # If users try to use ttcopy with stdin for the first time.
+    if [ ! -t 0 ]; then
+        echo "Standard input is discarded because credentials are not set yet." >&2
+        echo "Please try again after setting ID/Password." >&2
+        echo >&2
+        cat /dev/stdin > /dev/null
+    fi
+
+    echo "Set default ID/Password." >&2
+    # Following messages are similar to ssh-keygen's one.
+
+    printf "Enter ID for ttcopy/ttpaste:" >&2
```

Comment:
Putting a space after the colon is cool 😎 

😕  `Enter ID for ttcopy/ttpaste:mecab`
😆    `Enter ID for ttcopy/ttpaste: mecab`